Sources/Networking/Modifiers/Processors/EndpointRequestStorageProcessor/EndpointRequestStorageProcessor.swift

```diff
@@ -137,8 +137,8 @@ private extension EndpointRequestStorageProcessor {
         _ response: Response,
         endpointRequest: EndpointRequest,
         urlRequest: URLRequest
-    ) {
-        Task(priority: .background) { [weak self] in
+    ) async {
+        await Task(priority: .background) { [weak self] in
```

Comment:
Please check those changes @cejanen @matejmolnar

Reply:
I don't think we want to do this. It was specifically written to execute in a background task in order to not delay requests processing in the app. Which actually reminds me that the Task should also be detached.

Reply:
Quick question before I try different solutions how to deterministically write tests without using the sleeps. 

Was this really an issue? That the Filemanager/Encoder was slow? 

And we are talking about debug/test builds only right? This isnt supposed to be used in production.

Reply:
1) this processor should be used only in debug 
2) even in debug we didn't want to block response processing just by storing the response to have mock data available, therefore the task was not awaiting
3) I don't like the sleep as well but if I remember correctly if the test was run immediately the file was not stored yet (some delay there is)

Reply:
@cejanen 3. not sure if understand this point entirely 😅 . The reason why I did those changes is to know when this `Task` finished (so that I know in the test), because it is run on a BG thread.

Reply:
I am not aware of a way to test the background task write operations on FileManger deterministically. But I suppose we could add some observation logic to the newly added MockFileManager 🤔

Reply:
Done 👍, please have a look. The `FileManager` can't observe files and it probably wouldn't help because the final data writing was done via `Data.write`.

Reply:
Oh right, I didn't realise that 😅 I like the new implementation 🙂

Reply:
> @cejanen 3. not sure if understand this point entirely 😅 . The reason why I did those changes is to know when this `Task` finished (so that I know in the test), because it is run on a BG thread.

I understand why you did it. Only saying we don't want it :-D For test convenience you changed the behaviour of that feature which is inconvenient 🦜

Reply:
@cejanen yep, got it. Reworked 👍